Mathlib/SetTheory/Ordinal/Arithmetic.lean

```diff
@@ -332,16 +332,26 @@ theorem bounded_singleton {r : α → α → Prop} [IsWellOrder α r] (hr : (typ
   rw [@enum_lt_enum _ r, Subtype.mk_lt_mk]
   apply lt_succ
 
--- Porting note: `· < ·` requires a type ascription for an `IsWellOrder` instance.
-theorem type_subrel_lt (o : Ordinal.{u}) :
-    type (@Subrel Ordinal (· < ·) { o' : Ordinal | o' < o }) = Ordinal.lift.{u + 1} o := by
+theorem type_Iio_ordinal (o : Ordinal.{u}) :
+    type (@Subrel Ordinal (· < ·) (Iio o)) = Ordinal.lift.{u + 1} o := by
```

Comment:
Can the `@` be dropped here and/or below?

Reply:
No, although I'm not sure why (it seems to correctly fill the implicit argument with `Ordinal` but then fail to synth the well order).

Reply:
`type` and `typein` both need the explicit type argument when being used with any sort of polymorphic notation. I've thought about making a PR to turn them into explicit args but I haven't come around to that.